posthog/models/feature_flag.py

```diff
@@ -101,6 +100,41 @@ def get_filters(self):
             }
 
 
+# TODO assume that the seven types have enough in common that they will all use the same version model
+class ModelVersion(models.Model):
+    class Meta:
+        unique_together = ["model_name", "versioned_at"]
+
+    model_state = models.JSONField(null=False)
+    model_name = models.fields.TextField(null=False)
+    # to avoid an integer version field for ordering revisions
+    versioned_at: models.DateTimeField = models.DateTimeField(default=timezone.now)
+
+    # don't store a reference cos the team or user might get deleted.  ¯\_(ツ)_/¯
+    # audit log should be immutable
+    team_id = models.PositiveIntegerField(null=False)
+    created_by_name = models.TextField(null=False)
+    created_by_id = models.PositiveIntegerField(null=False)
+
+
+@receiver(post_save, sender=FeatureFlag, dispatch_uid="save_feature_flag_version")
+def save_version(sender, instance: FeatureFlag, **kwargs):
+    try:
+        version = ModelVersion(
+            #  ¯\_(ツ)_/¯ serialize the instance as a list and then chop off the square braces
```

Comment:
👀

Reply:
Took 60% of the time to write the example working through all the ways people claim you can serialize a model 🐰